### ISSUE TITLE
When I use the AR camera I can't use the method "GetImageFromGallery" to select the picture, and call back is null

### ISSUE THREAD

**Issue by RobinGitNa** [yasirkula/UnityNativeGallery#323]
After careful investigation, the problem was found in the NGMediaReceiveCallbackiOS class，
	private void Update()
		{
			if( IsBusy )
			{
				if( Time.realtimeSinceStartup >= nextBusyCheckTime )
				{
					nextBusyCheckTime = Time.realtimeSinceStartup + 1f;

					if( _NativeGallery_IsMediaPickerBusy() == 0 )
					{
						IsBusy = false;

						NativeGallery.MediaPickCallback _callback = callback;
						callback = null;

						NativeGallery.MediaPickMultipleCallback _callbackMultiple = callbackMultiple;
						callbackMultiple = null;

						if( _callback != null )
							_callback( null );

						if( _callbackMultiple != null )
							_callbackMultiple( null );
					}
				}
			}
		}

So I would like to know what the above method means, and will it have any effect after disabling it？Thank you so much!

**Comment by yasirkula** [comment]
It invokes the callback if the native image picker is silently closed. Can you describe the image picker UI's behaviour with and without this code?

**Comment by RobinGitNa** [comment]
with this code：When I use the AR camera I can't use the method "GetImageFromGallery" to select any picture, and call back is null，But I found that I can open the phone album normally, and there is a picture path back before update method runs;
without  this code : I can open the phone album normally, and there is a picture path back,and nothing else out of the ordinary.

**Comment by yasirkula** [comment]
Thank you for the elaboration.

> But I found that I can open the phone album normally, and there is a picture path back before update method runs

What I understand here is that the callback is invoked with a proper path before that Update method runs?

**Comment by RobinGitNa** [comment]
Yes, I'm sorry I'm not good at English.
This is because the condition "_NativeGallery_IsMediaPickerBusy() == 0" is always true after opening the AR camera。
In short, after the Update method is commented out, the image path can be obtained, and this phenomenon only occurs in ios, and the image path can be obtained even if the AR camera is opened in Android。

**Comment by yasirkula** [comment]
I couldn't give it a thought due to my current schedule. This is still in my mind.

**Comment by RobinGitNa** [comment]
> I couldn't give it a thought due to my current schedule. This is still in my mind.

Thank you very much

**Comment by yasirkula** [comment]
I'm trying to pinpoint the root of the issue. Were you testing on an iPad? If so, did the problem occur on an iPhone, as well?

**Comment by RobinGitNa** [comment]
I only tested it on an iPhone.

**Comment by yasirkula** [comment]
Can you put NSLogs inside this function to see how it returns 0: https://github.com/yasirkula/UnityNativeGallery/blob/66157bd32db65439aca72ad5ed0f0fb702d6ef72/Plugins/NativeGallery/iOS/NativeGallery.mm#L670-L699

For starters, you should log imagePickerState, whether or not imagePicker/imagePickerNew is nil, whether or not its presentingViewController is UnityGetGLViewController() (if not, what is it? Is it nil?)

**Comment by RobinGitNa** [comment]
sorry，I was very busy recently.  I ran some tests, but they didn't seem out of the ordinary, did they？
code：
+ (int)isMediaPickerBusy
{
    NSlog(@"imagePickerState:%d",imagePickerState);
	if( imagePickerState == 2 )
		return 1;
	
	NSlog(@"imagePicker != nil:%d",imagePicker != nil );
	if( imagePicker != nil )
	{
	NSlog(@"imagePicker :%@",imagePicker);
		if( imagePickerState == 1 || [imagePicker presentingViewController] == UnityGetGLViewController() )
			return 1;
		else
		{
			imagePicker = nil;
			return 0;
		}
	}
#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 140000
     NSlog(@"14.0--imagePickerState:%d",imagePickerState);
	 NSlog(@"14.0--imagePickerNew != nil:%d",imagePickerNew != nil);
	else if( CHECK_IOS_VERSION( @"14.0" ) && imagePickerNew != nil )
	{
	 NSlog(@"14.0--imagePickerNew:%@",imagePickerNew);
		if( imagePickerState == 1 || [imagePickerNew presentingViewController] == UnityGetGLViewController() )
			return 1;
		else
		{
			imagePickerNew = nil;
			return 0;
		}
	}
#endif
	else
		return 0;
}


test result：

imagePickerState:1
imagePicker != nil:0
14.0--imagePickerState:1
14.0--imagePickerNew != nil:1
14.0--imagePickerNew :<PHPickerViewController: 0x125b34490>
imagePickerState:0
imagePicker != nil:0
14.0--imagePickerState:0
14.0--imagePickerNew != nil:1
Picked an image
/var/mobile/Containers/Data/Application/CE9F5A69-F226-487D-808E-291FED04C802/Library/Caches/pickedMedia1.heic -- imgName :pickedMedia1_1719390587686.000.heic ---length:13698421 
UnityEngine.DebugLogHandler:Internal_Log(LogType, LogOption, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
ARPlace_ImageView:<LoadImgPanel>b__8_0(String)
NativeGalleryNamespace.NGMediaReceiveCallbackiOS:OnMediaReceived(String)

**Comment by yasirkula** [comment]
Yes it looks ordinary. I was suspecting that the imagePicker's presentingViewController check was somehow failing in AR but the logs say otherwise. I'll investigate the parts where imagePickerState is set to 0. Were you receiving the null callback immediately after invoking GetImageFromGallery or after picking an image from the gallery?

**Comment by RobinGitNa** [comment]
Yes，I received the null callback immediately  after picking an image from the gallery.
Sorry, I don't have a mac , I can't provide more information about the test.

**Comment by yasirkula** [comment]
Hello again. I'm trying to debug this issue once again. On AR, does the app pause/stop while the image picker is visible? If you get the chance, could you put Debug.Log(Time.frameCount) inside NGMediaReceiveCallbackiOS.Update and NGMediaReceiveCallbackiOS.OnMediaReceived so that we can understand which one is called when?
